### PR TITLE
Remove AES and SubtleCrypto from WASM

### LINKED ISSUE #73858: Remove AES and SubtleCrypto from WASM
We've concluded that using the `SharedArrayBuffer` approach via a worker is not yielding the end-to-end experience or quality necessary to release this integration. Details and commentary on dotnet/runtime#72810 show some specific challenges that emerged:

1. Blazor uses their own runtime initialization path and the crypto initialization call is not included in that path (or public)
    * This was addressable
2. Blazor's publish targets limit what files end up in the publish output history
    * This was addressable
3. Chromium is soon going to start limiting `SharedArrayBuffer` to be available only for cross-origin isolated sites. This will take effect as of Chromium v106.
    * This would require impactful changes to behavior that can't be justified at this stage of the release cycle for this feature
    * dotnet/aspnetcore#42114
4. A deadlock that was discovered
    * This specific deadlock was addressable, but it's possible others are lurking
    * dotnet/runtime#73537

These challenges lead to discussions about how often the managed implementations apply vs. the SubtleCrypto implementations. When we laid out the [plan to rely on SubtleCrypto through `SharedArrayBuffer`](https://github.com/dotnet/designs/blob/main/accepted/2021/blazor-wasm-crypto.md#discussion-1), we understood this would enable us to use SubtleCrypto in ~90% of scenarios. The plan was to only fall back to managed implementations "on browsers which do not support `SharedArrayBuffer`. However, with the recognition that the default end-to-end experience disallows use of `SharedArrayBuffer` broadly, that changed the picture dramatically. We now find ourselves in the situation where 90% or more of scenarios will be relying on the managed implementations, and customers desiring the native implementations must consider tradeoffs for other behavior changes that will occur in their applications. That's not an acceptable outcome.

With these findings and realizations, we're dialing back what we'll be able to support, and we will not rely on a `SharedArrayBuffer` bridge to invoke async SubtleCrypto APIs from synchronous .NET code. Instead, the supported algorithms will be exclusively supported via managed code and scoped to algorithms without known side channel leaks of the key. Per the original design document:

> The implementations of our in-box managed algorithms will be faithful to the appropriate standards. However, we will not pursue FIPS 140-2 or any other certification for them. Our in-box managed algorithm implementations are not intended to be free of side channels.

Here's the net result of the changes to what will be supported.

| API | .NET 6 | .NET 7 - Planned | .NET 7 - Actual |
|-|-|-|-|
| MD5 | Not Supported | Not supported | Not supported |
| **SHA1** | **Managed** | **SubtleCrypto (managed fallback)** | **Managed** |
| **SHA256** | **Managed** | **SubtleCrypto (managed fallback)** | **Managed** |
| **SHA384** | **Managed** | **SubtleCrypto (managed fallback)** | **Managed** |
| **SHA512** | **Managed** | **SubtleCrypto (managed fallback)** | **Managed** |
| HMACMD5 | Not supported | Not supported | Not supported |
| **HMACSHA1** | **Not supported** | **SubtleCrypto (managed fallback)** | **Managed** |
| **HMACSHA256** | **Not supported** | **SubtleCrypto (managed fallback)** | **Managed** |
| **HMACSHA384** | **Not supported** | **SubtleCrypto (managed fallback)** | **Managed** |
| **HMACSHA512** | **Not supported** | **SubtleCrypto (managed fallback)** | **Managed** |
| **AES-CBC (key sizes 128 + 256)** | **Not supported** | **SubtleCrypto (managed fallback)** | **Not supported** |
| **PBKDF2 (via Rfc2898DeriveBytes)** | **Not supported** | **SubtleCrypto (managed fallback)** | **Managed** |
| **HKDF** | Not supported | **SubtleCrypto (managed fallback)** | **Managed** |
| Asymmetric | Not supported | Not supported | Not supported |
| Certificates | Not supported | Not supported | Not supported |

We do not have plans to try this integration with SubtleCrypto again. Instead, for customers needing native implementations or unsupported algorithms such as AES, we advise they use JavaScript interop to invoke SubtleCrypto directly. That approach will allow the application to be written such that the call can be made in an asynchronous context.

### Tasks

- [x] Remove the worker from WasmAppBuilder
- [x] Remove the call from startup.ts
- [x] Remove the DLLImports
- [x] Remove the C# code that is calling this
- [x] Remove the special `js-module-crypto` in `AssetBehaviours` typescrypt
- [ ] Remove `js-module-crypto` handling in the SDK (replace it with `js-module-threads` would be useful)
- [x] Remove managed AES implementation
- [x] Remove the subtlecrypto run of the tests (https://github.com/dotnet/runtime/pull/73884)

cc @pavelsavara @maraf @lambdageek @eerhardt

### PR DESCRIPTION
**Pull request by eerhardt**
Fix #73858

Tasks from #73858

- [x] Remove the worker from WasmAppBuilder
- [x] Remove the call from startup.ts
- [x] Remove the DLLImports
- [x] Remove the C# code that is calling this
- [x] Remove the special `js-module-crypto` in `AssetBehaviours` typescrypt
- [ ] Remove `js-module-crypto` handling in the SDK (replace it with `js-module-threads` would be useful)
- [x] Remove managed AES implementation

### CODE REVIEW

File: src/tasks/WasmAppBuilder/WasmAppBuilder.cs [hunk 1]
```diff
@@ -320,7 +320,6 @@ private bool ExecuteInternal ()
 
         config.Assets.Add(new VfsEntry ("dotnet.timezones.blat") { VirtualPath = "/usr/share/zoneinfo/"});
         config.Assets.Add(new WasmEntry ("dotnet.wasm") );
-        config.Assets.Add(new CryptoWorkerEntry ("dotnet-crypto-worker.js") );
```
Comment (lambdageek) [inline src/tasks/WasmAppBuilder/WasmAppBuilder.cs:323]: should we also remove the definition of the `CryptoWorkerEntry` class?

**Review by lambdageek — COMMENTED** [review]
src/mono and src/tasks changes lgtm

**Review by bartonjs — APPROVED** [review]

File: src/tasks/WasmAppBuilder/WasmAppBuilder.cs (hunk 1 above)
Comment (eerhardt) [reply src/tasks/WasmAppBuilder/WasmAppBuilder.cs:323]: Yep! Missed that one.

**Review by pavelsavara — APPROVED** [review]
TypeScript changes LGTM, I didn't check the rest.

**Pull request merged**